framework/include/utils/MathFVUtils.h

```diff
@@ -98,12 +110,34 @@ typename libMesh::CompareTypes<T, T2>::supertype
 linearInterpolation(const T & value1,
                     const T2 & value2,
                     const FaceInfo & fi,
-                    const bool one_is_elem)
+                    const bool one_is_elem,
+                    const InterpMethod interp_method = InterpMethod::Average)
 {
-  const auto coeffs = interpCoeffs(InterpMethod::Average, fi, one_is_elem);
+  const auto coeffs = interpCoeffs(interp_method, fi, one_is_elem);
   return coeffs.first * value1 + coeffs.second * value2;
```

Comment:
The function above it: `std::pair<Real, Real>interpCoeffs(...)` has a `mooseError` which gets triggered if interp_method is anything besides `average/skew-corrected/upwind`. We will not hit upwind with the new addition of the code since the parameter reads MooseEnums of `average/skew-corrected/vertex-based`. In case the input is vertex-based, we get an error. However, it is impossible to get an error here since we use Average interpolation for the face gradient even in cases when the face values are interpolated using the vertex-based stencil. See the additional comment below for more details.

Reply:
I usually assume that future developers may come along and misuse code, even if no code misuses it now. Adding assertions doesn’t cost us any run time so I think it’s good to add assertions whenever they make sense. I’ll let you decide if an assertion makes sense here

Reply:
Added an assert to the function.